### ISSUE TITLE
Question: the libhugetlbfs does not work properly on Ubuntu 22.04 (6.8.0 kernel). Do i miss something?

### ISSUE THREAD

**Issue by Jingzhao123** [libhugetlbfs/libhugetlbfs#95]
Hi, everyone
I test a simple code on my desktop. But, i found the program can not work on hugepage mem. And i use the same code on another environment(ubuntu 20.04, 5.4.0). It is OK. **Did i miss something when configuring the hugepage**?

- Ubuntu 22.04:

```
hugepage# LD_PRELOAD=libhugetlbfs.so HUGETLB_MORECORE=yes ./a.out 
libhugetlbfs [host]: INFO: Found pagesize 2048 kB
libhugetlbfs [host]: ERROR: Line too long when parsing mounts
libhugetlbfs [host]: INFO: Detected page sizes:
libhugetlbfs [host]: INFO:    Size: 2048 kB (default)  Mount: /dev/hugepages
libhugetlbfs [host]: INFO: Parsed kernel version: [6] . [8] . [0] 
libhugetlbfs [host]: INFO: Feature private_reservations is present in this kernel
libhugetlbfs [host]: INFO: Feature noreserve_safe is present in this kernel
libhugetlbfs [host]: INFO: Feature map_hugetlb is present in this kernel
libhugetlbfs [host]: INFO: Kernel has MAP_PRIVATE reservations.  Disabling heap prefaulting.
libhugetlbfs [host]: INFO: Kernel supports MAP_HUGETLB
libhugetlbfs [host]: INFO: HUGETLB_SHARE=0, sharing disabled
libhugetlbfs [host]: INFO: HUGETLB_NO_RESERVE=no, reservations enabled
libhugetlbfs [host]: INFO: No segments were appropriate for remapping
libhugetlbfs [host]: INFO: setup_morecore(): heapaddr = 0x5de35d000000
```

Info:

```
hugepage# hugeadm --list-all-mounts
libhugetlbfs: ERROR: Line too long when parsing mounts
hugeadm:DEBUG: HUGETLB_VERBOSE='4'
Mount Point          Options
/dev/hugepages       rw,relatime,pagesize=2M

hugepage# cat /proc/meminfo |grep Huge
AnonHugePages:     26624 kB
ShmemHugePages:        0 kB
FileHugePages:         0 kB
HugePages_Total:    1000
HugePages_Free:     1000
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:         2048000 kB
```

- Ubuntu 20.04:

```
hugepage$ LD_PRELOAD=libhugetlbfs.so HUGETLB_MORECORE=yes  ./a.out
libhugetlbfs [host2]: INFO: Found pagesize 2048 kB
libhugetlbfs [host2]: INFO: Detected page sizes:
libhugetlbfs [host2]: INFO:    Size: 2048 kB (default)  Mount:
libhugetlbfs [host2]: INFO: Parsed kernel version: [5] . [4] . [0]
libhugetlbfs [host2]: INFO: Feature private_reservations is present in this kernel
libhugetlbfs [host2]: INFO: Feature noreserve_safe is present in this kernel
libhugetlbfs [host2]: INFO: Feature map_hugetlb is present in this kernel
libhugetlbfs [host2]: INFO: Kernel has MAP_PRIVATE reservations.  Disabling heap prefaulting.
libhugetlbfs [host2]: INFO: Kernel supports MAP_HUGETLB
libhugetlbfs [host2]: INFO: HUGETLB_SHARE=0, sharing disabled
libhugetlbfs [host2]: INFO: HUGETLB_NO_RESERVE=no, reservations enabled
libhugetlbfs [host2]: INFO: No segments were appropriate for remapping
libhugetlbfs [host2]: INFO: setup_morecore(): heapaddr = 0x559ec8e00000
libhugetlbfs [host2]: INFO: hugetlbfs_morecore(1052672) = ...
libhugetlbfs [host2]: INFO: heapbase = 0x559ec8e00000, heaptop = 0x559ec8e00000, mapsize = 0, delta=1052672
libhugetlbfs [host2]: INFO: Attempting to map 2097152 bytes
libhugetlbfs [host2]: INFO: ... = 0x559ec8e00000
libhugetlbfs [host2]: INFO: hugetlbfs_morecore(0) = ...
libhugetlbfs [host2]: INFO: heapbase = 0x559ec8e00000, heaptop = 0x559ec8f01000, mapsize = 200000, delta=-1044480
libhugetlbfs [host2]: INFO: ... = 0x559ec8f01000
libhugetlbfs [host2]: INFO: hugetlbfs_morecore(54525952) = ...
libhugetlbfs [host2]: INFO: heapbase = 0x559ec8e00000, heaptop = 0x559ec8f01000, mapsize = 200000, delta=53481472
libhugetlbfs [host2]: INFO: Attempting to map 54525952 bytes
libhugetlbfs [host2]: INFO: ... = 0x559ec8f01000
```

**Comment by shivankgarg98** [comment]
Ubuntu 22.04 has newer glibc. Read below ticket:
https://github.com/libhugetlbfs/libhugetlbfs/issues/52

TLDR: export GLIBC_TUNABLES=glibc.malloc.hugetlb=2

**Comment by Jingzhao123** [comment]
Great. Thanks.
Now, it works.
